src/I18n/DateFormatTrait.php

```diff
@@ -48,6 +48,22 @@ trait DateFormatTrait
      */
     protected static $_formatters = [];
 
+    /**
+     * The format to use when when converting this object to json
+     *
+     * The format should be either the formatting constants from IntlDateFormatter as
+     * described in (https://secure.php.net/manual/en/class.intldateformatter.php) or a pattern
+     * as specified in (http://www.icu-project.org/apiref/icu4c/classSimpleDateFormat.html#details)
+     *
+     * It is possible to provide an array of 2 constants. In this case, the first position
+     * will be used for formatting the date part of the object and the second position
+     * will be used to format the time part.
+     *
+     * @var string|array|int|callable
```

Comment:
Can we limit it to just `Closure` instead of `callable`?

Reply:
Sure.